client/features/settings-page/Settings.tsx

```diff
@@ -1,9 +1,118 @@
-import React from 'react';
-import useDocumentTitle from '@hooks/useDocumentTitle';
+import React, { useEffect } from 'react';
+import { Controller, useForm } from 'react-hook-form';
+import { Box, Typography, TextField, Button } from '@mui/material';
+import styles from '@styles/SettingsPage.module.css';
+import {
+  useFetchUserProfile,
+  useUpdateUserProfile,
+} from '@hooks/useUserCrudOperations';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { userDisplayNameSchema } from '@validators/validateUserDisplayName';
+import { useQueryClient } from '@tanstack/react-query';
+import useDebounce from '@hooks/useDebounce';
+import { AxiosError } from 'axios';
+import { useNotification } from '@hooks/useNotification';
+import Loader from '@components/common/Loader';
 
-const Settings = () => {
-  useDocumentTitle('Settings');
-  return <div>Settings Placeholder</div>;
+interface UserProfileFormData {
+  name: string;
+}
+
+const Settings: React.FC = () => {
+  const { showNotification } = useNotification();
+
+  const queryClient = useQueryClient();
+  const { data: userProfile, isFetching } =
```

Comment:
Will need `fetchingStatus` prop to get the fetching state. [Doc](https://tanstack.com/query/v4/docs/framework/react/guides/queries)